staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -261,6 +261,7 @@ func patchResource(
 			// 3. ensure no conflicts between the two patches
 			// 4. apply the #1 patch to the currentJS object
 
+			currentGVK := currentObject.GetObjectKind().GroupVersionKind()
```

Comment:
Can you add this for the [strategic merge patch condition](https://github.com/kubernetes/kubernetes/blob/6659f2a7d894a404d59265a708a2d2bd03806e98/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go#L215) too?

This will fix the wrong apiVersion issue:

```sh
$ kubectl patch crontab my-new-cron-object -p '{}'

# before this change
The CronTab "my-new-cron-object" is invalid: apiVersion: Invalid value: "stable.example.com/__internal": must be stable.example.com/v1

# after this change
crontab "my-new-cron-object" not patched

Reply:
Not very sure about this.

I found that we can only put an empty strategic merge patch to crd.

Otherwise we get:
```
$ kubectl patch coredump coredump-sleep-counter-foo -p '{"status":{"message1":"2222221"}}'
Error from server: unable to find api field in struct Unstructured for the json field "status"
```

Does this change to strategic make real enhancement?

Reply:
> Does this change to strategic make real enhancement?

Not really. It won't make any change after https://github.com/kubernetes/kubernetes/pull/53558. :+1: